js/base/Exchange.js

```diff
@@ -604,7 +574,30 @@ module.exports = class Exchange {
                 ].join (', ') + ')'
             }
         }
-        throw new ErrorClass ([ this.id, method, url, code, reason, details ].join (' '))
+        if (ErrorClass) {
```

Comment:
`if (ErrorClass)` is always true in this case, do we really need this check?
Line 582 duplicates line 559.